### DISCUSSION TITLE
Update ramanchada2

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#66] [Q&A] [answered]
What would be the proper way to update ramanchada2 from github using the terminal?

**Comment by georgievgeorgi** [comment, marked as answer]
## TL;DR: `git pull`

## There are multiple aspects.
### User's local changes
The user needs to make sure all local changes made to git-tracked files are [committed](https://git-scm.com/docs/git-commit), [stashed](https://git-scm.com/docs/git-stash) or preserved in other way. If local changes are no longer needed one can reset HEAD to the last state by [`git reset --hard`](https://git-scm.com/docs/git-reset). `--hard` option should be used with caution! It discards all local changes that are not commited

### Obtain last version from git
Once the working directory is clear, the last version can be obtained by:
```
git checkout main
git pull
```

### Python environment
Python packages can be installed as regular packages or in editable mode. Usually preferred method for development is the editable mode -- [`pip install --editable .`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-e) ran from the ramanchada2 directory.  In this mode changes to local files lead to immediate changes to the library -- no reinstallation is needed.

### Jupyter kernels
Once a Jupyter kernel is configured with the virtual environment, no other changes are needed. Just restart the kernel and run the notebook cells again.
